src/sas/sasgui/perspectives/fitting/fitpage.py

```diff
@@ -3181,11 +3236,17 @@ def _set_smear_buttons(self):
         if self.data.__class__.__name__ == "Data2D" or \
                     self.enable2D:
             self.slit_smearer.Disable()
-            self.pinhole_smearer.Enable(True)
             self.default_mask = copy.deepcopy(self.data.mask)
+            if self.model is not None:
+                self.pinhole_smearer.Enable(True)
+
+        elif self.data.__class__.__name__ == "Data1D":
```

Comment:
Here's another place for an `isinstance` call.

Reply:
yes ... many places that nasty thing is .... if I can get isinstance to work..